### ISSUE TITLE
{{ pathto('index') }} in some theme layout.html.

### ISSUE THREAD

**Issue by noboruatkek** [sphinx-doc/sphinx#11663]
### Describe the bug

In the themes, haiku and scrolls, each layout.html uses "{{ pathto('index') }}".   Some links in the generated html output does not work as expected, when user setup "master_doc" nor "rood_doc" to anything other than "index".

It should be {{ pathto(root_doc) }}" , instead. 

### How to Reproduce

set "root_doc" to any value to "contents", for example. ( root file should be contents.rst as well.)

### Environment Information

```text
Platform:              darwin; (macOS-14.0-arm64-arm-64bit)
Python version:        3.9.13 (v3.9.13:6de2ca5339, May 17 2022, 11:37:23) 
[Clang 13.0.0 (clang-1300.0.29.30)])
Python implementation: CPython
Sphinx version:        7.2.2
Docutils version:      0.20.1
Jinja2 version:        3.1.2
Pygments version:      2.15.1
```

**Comment by AA-Turner** [comment]
PR welcome.

A

**Comment by AA-Turner** [comment]
Fixed in #11801.

A
